webapp/tests/karma/ts/modules/tasks/tasks.component.spec.ts

```diff
@@ -0,0 +1,227 @@
+      getComponent();
+    });
+    expect(component.loading).to.be.false;
+    expect(!!component.hasTasks).to.be.false;
+    expect(!!component.error).to.be.false;
+    expect(component.tasksDisabled).to.be.true;
+  });
+
+  it('rules engine throws in initialization', async () => {
+    RulesEngine.isEnabled.rejects('error');
+    await new Promise(resolve => {
+      sinon.stub(TasksActions.prototype, 'setTasksList').callsFake(resolve);
+      getComponent();
+    });
+    expect(component.loading).to.be.false;
+    expect(!!component.hasTasks).to.be.false;
+    expect(component.error).to.be.true;
+    expect(!!component.tasksDisabled).to.be.false;
+    expect((<any>TasksActions.prototype.setTasksList).args).to.deep.eq([[[]]]);
+  });
+
+  it('tasks render', async () => {
+    const now = moment('2020-10-20');
+    const futureDate = now.clone().add(3, 'days');
+    const pastDate = now.clone().subtract(3, 'days');
+    clock = sinon.useFakeTimers(now.valueOf());
+    const taskDocs = [
+      { _id: '1', emission: { _id: 'e1', dueDate: futureDate.format('YYYY-MM-DD') }},
+      { _id: '2', emission: { _id: 'e2', dueDate: pastDate.format('YYYY-MM-DD') }},
+    ];
+    RulesEngine.fetchTaskDocsForAllContacts.resolves(taskDocs);
+
+    await new Promise(resolve => {
+      sinon.stub(TasksActions.prototype, 'setTasksList').callsFake(resolve);
+      getComponent();
+    });
+    expect(component.loading).to.be.false;
+    expect(component.tasksDisabled).to.be.false;
+    expect(component.hasTasks).to.be.true;
+    expect(!!component.error).to.be.false;
+    const expectedTasks = [
+      { _id: 'e1', dueDate: futureDate.format('YYYY-MM-DD'), overdue: false, date: new Date(futureDate.valueOf()) },
+      { _id: 'e2', dueDate: pastDate.format('YYYY-MM-DD'), overdue: true, date: new Date(pastDate.valueOf()) },
+    ];
+    expect((<any>TasksActions.prototype.setTasksList).args).to.deep.eq([[expectedTasks]]);
+  });
+
+  it('rules engine yields no tasks', async () => {
+    await new Promise(resolve => {
+      sinon.stub(TasksActions.prototype, 'setTasksList').callsFake(resolve);
+      getComponent();
+    });
+    expect(component.loading).to.be.false;
+    expect(component.tasksDisabled).to.be.false;
+    expect(component.hasTasks).to.be.false;
+    expect(!!component.error).to.be.false;
+    expect(RulesEngine.fetchTaskDocsForAllContacts.callCount).to.eq(1);
+    expect((<any>TasksActions.prototype.setTasksList).args).to.deep.eq([[[]]]);
+  });
+
+  it('changes feed', async () => {
+    contactTypesService.includes
+      .withArgs(sinon.match({ type: 'person' })).returns(true)
+      .withArgs(sinon.match({ type: 'clinic' })).returns(true)
+      .withArgs(sinon.match({ type: 'contact' })).returns(true);
+    await new Promise(resolve => {
+      sinon.stub(TasksActions.prototype, 'setTasksList').callsFake(resolve);
+      getComponent();
+    });
+
+    const changesFeed = Changes.args[0][0];
```

Comment:
I got a bit confused when I saw this: "isn't it that changes service needs the subscribe function" then I got back the the test module definition and saw the thing 😅

Reply:
Yea, I'll update them since I couldn't get Git to play nice with history.